engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/source/SpecificRecordKafkaAvroSourceFactory.scala

```diff
@@ -0,0 +1,31 @@
+package pl.touk.nussknacker.engine.avro.source
+
+import javax.validation.constraints.NotBlank
+import org.apache.avro.specific.SpecificRecord
+import org.apache.flink.streaming.api.functions.TimestampAssigner
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.NodeId
+import pl.touk.nussknacker.engine.api.editor.{SimpleEditor, SimpleEditorType}
+import pl.touk.nussknacker.engine.api.process.ProcessObjectDependencies
+import pl.touk.nussknacker.engine.api.typed.ReturningType
+import pl.touk.nussknacker.engine.api.{MetaData, MethodToInvoke, ParamName}
+import pl.touk.nussknacker.engine.avro.schemaregistry.{SchemaRegistryProvider, SpecificRecordEmbeddedSchemaDeterminer}
+import pl.touk.nussknacker.engine.kafka.KafkaFactory._
+import pl.touk.nussknacker.engine.kafka.source.KafkaSource
+import pl.touk.nussknacker.engine.kafka.{KafkaConfig, KafkaUtils}
+
+import scala.reflect._
+
+class SpecificRecordKafkaAvroSourceFactory[T <: SpecificRecord: ClassTag](schemaRegistryProvider: SchemaRegistryProvider[T], processObjectDependencies: ProcessObjectDependencies, timestampAssigner: Option[TimestampAssigner[T]])
```

Comment:
Why we don't have SpecificKafkaAvroSinkFactory?

Reply:
On sink side it is not needed to specify that it is either generic or specific case - current `KafkaAvroSinkFactory` handles both cases. One thing that could be useful for backward compatibility is sink factory without version param and with `schemaDeterminer` = `UsingRecordSchemaInRuntimeAvroSchemaDeterminer` but it is quite easy to achive with current api so I won't add it here